editor/src/core/model/get-uid-mappings.ts

```diff
@@ -0,0 +1,277 @@
+      break
+    case 'JS_ELEMENT_ACCESS':
+      extractUidFromJSXElementChild(working, workingDupliactedUIDs, filePath, element.onValue)
+      extractUidFromJSXElementChild(working, workingDupliactedUIDs, filePath, element.element)
+      break
+    default:
+      assertNever(element)
+  }
+}
+
+function extractUIDFromArbitraryBlock(
+  working: Set<string>,
+  workingDupliactedUIDs: DuplicateUIDsResult,
+  filePath: string,
+  arbitraryBlock: ArbitraryJSBlock,
+): void {
+  checkUID(working, workingDupliactedUIDs, filePath, arbitraryBlock.uid)
+  for (const elementWithin of Object.values(arbitraryBlock.elementsWithin)) {
+    extractUidFromJSXElementChild(working, workingDupliactedUIDs, filePath, elementWithin)
+  }
+}
+
+export function extractUIDFromTopLevelElement(
+  working: Set<string>,
+  workingDupliactedUIDs: DuplicateUIDsResult,
+  filePath: string,
+  topLevelElement: TopLevelElement,
+): void {
+  switch (topLevelElement.type) {
+    case 'UTOPIA_JSX_COMPONENT':
+      extractUidFromJSXElementChild(
+        working,
+        workingDupliactedUIDs,
+        filePath,
+        topLevelElement.rootElement,
+      )
+      if (topLevelElement.arbitraryJSBlock != null) {
+        extractUIDFromArbitraryBlock(
+          working,
+          workingDupliactedUIDs,
+          filePath,
+          topLevelElement.arbitraryJSBlock,
+        )
+      }
+      break
+    case 'ARBITRARY_JS_BLOCK':
+      extractUIDFromArbitraryBlock(working, workingDupliactedUIDs, filePath, topLevelElement)
+      break
+    case 'IMPORT_STATEMENT':
+      break
+    case 'UNPARSED_CODE':
+      break
+    default:
+      assertNever(topLevelElement)
+  }
+}
+
+type FileToUidMapping = Map<string, Set<string>>
+
+let CachedUidsPerFile = new WeakMap<ParseSuccess, GetAllUniqueUidsResultForFile>()
+
+function collectUidsForFile(
+  filePath: string,
+  parseSuccess: ParseSuccess,
+): GetAllUniqueUidsResultForFile {
+  const cachedResult = CachedUidsPerFile.get(parseSuccess)
+  if (cachedResult != null) {
+    return cachedResult
+  }
+
+  let workingUniqueIDSet = emptySet<string>()
+  let workingDupliactedUIDs: DuplicateUIDsResult = new Map()
+  fastForEach(parseSuccess.topLevelElements, (tle) => {
+    extractUIDFromTopLevelElement(workingUniqueIDSet, workingDupliactedUIDs, filePath, tle)
+  })
+  const result = { uniqueIDs: workingUniqueIDSet, duplicateIDs: workingDupliactedUIDs }
+  CachedUidsPerFile.set(parseSuccess, result)
+  return result
+}
+
+function collectUidsForEachFile(projectContents: ProjectContentTreeRoot): GetAllUniqueUIDsResult {
+  let workingFileToUidMapping: FileToUidMapping = new Map()
+  let workingDupliactedUIDs: DuplicateUIDsResult = new Map()
+  walkContentsTreeForParseSuccess(projectContents, (filePath, parseSuccess) => {
+    const mappingsForFile = collectUidsForFile(filePath, parseSuccess)
+    // mappingsForFile.uniqueIDs.forEach((uid) => {
+    //   workingAllUids.add(uid)
+    // })
```

Comment:
Did you intend to leave this commented out code here?

Reply:
oops,fixed!